js/src/framework.ts

```diff
@@ -220,27 +218,25 @@ export async function runEvaluator(
           const meta = (o: Record<string, unknown>) =>
             (metadata = { ...metadata, ...o });
 
-          await traced(
-            async () => {
-              const outputResult = evaluator.task(datum.input, {
-                meta,
-                span: currentSpan(),
-              });
-              if (outputResult instanceof Promise) {
-                output = await outputResult;
-              } else {
-                output = outputResult;
-              }
-              currentSpan().log({ input: datum.input, output });
-            },
-            { name: "task" }
-          );
+          await currentSpan().traced("task", async () => {
```

Comment:
Per discussion RE deno issues, I think this will fail if async local storage doesn't work. Can we thread the relevant `span` here explicitly?